### DISCUSSION TITLE
Investigation of the Origin of the 2pi jumps

### DISCUSSION THREAD

**Discussion by mmikhasenko** [KaiHabermann/decayangle#49] [General]
## Idea

We start with the four-momentum configuration in CM frame using tau_3, i.e. `(phi_3, theta_3, phi_12)`.
Using the `helicity_angle` call other variables are computed, for example

`(phi_1, theta_1, phi_23)`

We know that in the _wigner angles_ of topology `(23)1` with respect to topology `(12)3` there are 2pi jumps.
Our claim is that the jump appears because of crossing 2pi borders in helicity angles `(phi_1, theta_1, phi_23)`

## Setup

is the same as for the previous exercise that shows the jump, i.e. `phi_3` is fixed to 0.7 (see `make_four_vectors(0.7, X, Y)` in [notebooks/three_body_angles.ipynb](https://github.com/KaiHabermann/decayangle/blob/master/notebooks/three_body_angles.ipynb))
 and `(theta_3, phi_12)` are varied.

We pick a 1d curve on theta_3, phi_12, and plot how it goes in topology `(23)1` angles
We need to make sure that the 1d curve crosses the `2pi flip regions`.

It can be a circle, but with large radius to cross the top right corner area.
```
theta_3, phi_12 = real(x), 2*imag(x)
```
where `x = pi/2+1.4*exp(i*pi*t)`, and we change t in the range [-0.5, 1.5].

## Angle Naming

We need to get the names of angles straight for clarity in the discussion. There are 4 helicity angles, and three Wigner angles  for every topology
let me use same indexing as in DPD paper.

For chains (12)3, the angles are

phi_3, theta_3, are of the subsystem in the CMF,
phi_12, theta_12 are helicity angles of resonance decay (particle 1) in the (12) rest frame.

It's understood that boost and Rz(phi_12) commute, therefore,
(phi_3, theta_3, phi_12) can be all viewed as the CMF angles, these are the Euler angles (RzRyRz) that determine the decay plane location

**Comment by KaiHabermann** [comment]
https://github.com/KaiHabermann/decayangle/pull/51

Here is a pull request with the extended notebook. then we have code to reference to

**Comment by KaiHabermann** [comment]
![output](https://github.com/user-attachments/assets/9ac17d4a-4fb4-45a8-b584-693f6dd11e4e)

Even though this plot is quite chaotic, one can nicely see the jumps in the helicity angles
